src/pocketmine/entity/Living.php

```diff
@@ -132,7 +132,18 @@ public function attack($damage, EntityDamageEvent $source){
 		$this->attackTime = 10; //0.5 seconds cooldown
 	}
 
-	public function knockBack(Entity $attacker, $damage, $x, $z, $base = 0.4){
+	/**
+	 * WARNING: Do not pass more arguments to this function than needed.
+	 * @param number $x
+	 * @param number $z
+	 * @param float  $base default 0.4
+	 */
+	public function knockBack($x, $z, $base = 0.4){
```

Comment:
Note: This really is BC compatible because of the func_get_args below. The warning message should be removed, however. Rather put a comment on the BC part saying why that exists there for future maintainers.